File: leetcode/python/1456_maximum_number_of_vowels_in_substring_of_given_length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        res = 0
        voewls = set(['a', 'e', 'i', 'o', 'u'])
        for c in s[:min(len(s), k)]:
            if c in voewls:
                res += 1
        count = res
        for i in range(k, len(s)):
            if s[i] in voewls:
                count += 1
            if s[i - k] in voewls:
                count -= 1
            res = max(res, count)
        return res

File: leetcode/rust/1456_maximum_number_of_vowels_in_substring_of_given_length.rs
use std::collections::HashSet;

impl Solution {
    pub fn max_vowels(s: String, k: i32) -> i32 {
        let mut res = 0i32;
        let voewls = HashSet::from(['a', 'e', 'i', 'o', 'u']);
        let chars = s.chars();

        for i in 0..k {
            if voewls.contains(&chars.nth(i as usize).unwrap()) {
                res += 1;
            }
        }
    }
}